### PR TITLE
Convert API doc schemas to YAML

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR converts several existing API doc schemas to YAML, which is easier to read and write and tends to be the standard for representing Swagger/OpenAPI. Several files were also renamed to match convention.

The only remaining JSON spec is annotation, which merits a separate task as it is a) complex and b) inaccurate.

In this PR, the _only_ meaningful change is that the previous version of the user schema allowed two types for `userid` in Profile responses: `string` and `null`. AFAIK we never return a `null` value for that field in reality, these days—every user has a `userid`, right? Alert me if I'm wrong. Otherwise the rendered output is data-identical.

This was just a quick end-of-the-week housekeeping task; there's certainly more docs organization to come. Ahhhh, YAML is just so much nicer to look at than JSON, in this context...

### CODE REVIEW

**Comment by robertknight** [comment]
> In this PR, the only meaningful change is that the previous version of the user schema allowed two types for userid in Profile responses: string and null. AFAIK we never return a null value for that field in reality, these days—every user has a userid, right? 

The `/api/profile` endpoint can be called without authentication, and the client does this if you're not logged in. In that case the `userid` field is `null`.

It is possible that we could remove support for unauthenticated requests to this endpoint in future, and require clients to get feature flags another way. The client could determine its log-in status based on whether it had an auth token or not.

File: docs/_extra/api-reference/schemas/user-new.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+User:
+  type: object
+  anyOf:
+    - required:
```
Comment (robertknight) [inline docs/_extra/api-reference/schemas/user-new.yaml]: Does this mean that you need to provide (authority, username, email) OR (authority, username, identities)?

Comment (robertknight) [reply docs/_extra/api-reference/schemas/user-new.yaml]: Yes, it seems so:

screenshot 2019-02-07 17 52 05

Is it possible to display a more helpful name in the "Any of" list in that view?

Comment (lyzadanger) [reply docs/_extra/api-reference/schemas/user-new.yaml]: It's JSON Schema terminology; maybe not the most immediately intuitive to everyone, but technically accurate.

**Comment by lyzadanger** [comment]
> > In this PR, the only meaningful change is that the previous version of the user schema allowed two types for userid in Profile responses: string and null. AFAIK we never return a null value for that field in reality, these days—every user has a userid, right?
> 
> The `/api/profile` endpoint can be called without authentication, and the client does this if you're not logged in. In that case the `userid` field is `null`.
> 
> It is possible that we could remove support for unauthenticated requests to this endpoint in future, and require clients to get feature flags another way. The client could determine its log-in status based on whether it had an auth token or not.

Ah! Good catch. I'll update the docs to re-add the `null` and also to explain the situation.

**Pull request merged**